main.go

```diff
@@ -117,30 +122,134 @@ type ObjRef struct {
 	Offloaded uint
 }
 
-// updateAutoretrieveIndex ticks every tickInterval and checks for new information to add to autoretrieve
-// If so, it updates the filecoin index with the new CIDs, saying they are present on autoretrieve
+// updateAutoretrieveIndex ticks every tickInterval and checks for new CIDs added
+// It updates storetheindex with the new CIDs, saying they are present on autoretrieve
 // With that, clients using bitswap can query autoretrieve servers using bitswap and get data from estuary
 func (s *Server) updateAutoretrieveIndex(tickInterval time.Duration, quit chan struct{}) error {
-	var autoretrieves []Autoretrieve
+	var autoretrieves []autoretrieve.Autoretrieve
 	var lastTickTime time.Time
-	ticker := time.NewTicker(tickInterval)
+	var curTime time.Time
+	var newContextID []byte
+	var err error
+
+	// Create index-provider engine (s.Node.IndexProvider) to send announcements to
+	// this needs to keep running continuously because storetheindex
+	// will come to fetch for advertisements "when it feels like it"
+	s.Node.ArEngine, err = autoretrieve.NewAutoretrieveEngine()
+	if err != nil {
+		return err
+	}
 
+	s.Node.ArEngine.RegisterMultihashLister(func(ctx context.Context, contextID []byte) (provider.MultihashIterator, error) {
+		var newContents []Content
```

Comment:
nitpick: it is ok to move these variables closer to select queries, no need to have them here since there is not they can be out of scope - easier to read